src/libsync/lockfilejobs.cpp

```diff
@@ -0,0 +1,199 @@
+        qCInfo(lcLockFileJob()) << "success" << path();
+        handleReply();
+        Q_EMIT finishedWithoutError();
+    }
+    return true;
+}
+
+void LockFileJob::setFileRecordLocked(SyncJournalFileRecord &record) const
+{
+    record._locked = (_lockStatus == SyncFileItem::LockStatus::LockedItem);
+    record._lockOwnerType = static_cast<int>(_lockOwnerType);
+    record._lockOwnerDisplayName = _userDisplayName;
+    record._lockOwnerId = _userId;
+    record._lockEditorApp = _editorName;
+    record._lockTime = _lockTime;
+    record._lockTimeout = _lockTimeout;
+}
+
+SyncJournalFileRecord LockFileJob::handleReply()
+{
+    const auto xml = reply()->readAll();
+
+    QXmlStreamReader reader(xml);
+
+    while (!reader.atEnd()) {
+        const auto type = reader.readNext();
+        const auto name = reader.name().toString();
+
+        switch (type) {
+        case QXmlStreamReader::TokenType::NoToken:
+        case QXmlStreamReader::TokenType::Invalid:
+        case QXmlStreamReader::TokenType::DTD:
+        case QXmlStreamReader::TokenType::EntityReference:
+        case QXmlStreamReader::TokenType::ProcessingInstruction:
+        case QXmlStreamReader::TokenType::Comment:
+        case QXmlStreamReader::TokenType::StartDocument:
+        case QXmlStreamReader::TokenType::Characters:
+        case QXmlStreamReader::TokenType::EndDocument:
+        case QXmlStreamReader::TokenType::EndElement:
+            break;
+        case QXmlStreamReader::TokenType::StartElement:
+            decodeStartElement(name, reader);
+            break;
+        }
+    }
+
+    SyncJournalFileRecord record;
+
+    if (_lockStatus == SyncFileItem::LockStatus::LockedItem) {
+        if (_lockOwnerType == SyncFileItem::LockOwnerType::UserLock && _userDisplayName.isEmpty()) {
+            return record;
+        }
+
+        if (_lockOwnerType == SyncFileItem::LockOwnerType::AppLock && _editorName.isEmpty()) {
+            return record;
+        }
+
+        if (_userId.isEmpty()) {
+            return record;
+        }
+
+        if (_lockTime <= 0) {
+            return record;
+        }
+
+        if (_lockTimeout <= 0) {
+            return record;
+        }
+    }
+
+    const auto relativePath = path().mid(1);
+    if (_journal->getFileRecord(relativePath, &record) && record.isValid()) {
+        setFileRecordLocked(record);
+        _journal->setFileRecord(record);
+        _journal->commit("lock file job");
+    }
+
+    return record;
+}
+
+void LockFileJob::decodeStartElement(const QString &name,
+                                     QXmlStreamReader &reader)
+{
+    if (name == QStringLiteral("lock")) {
+        const auto valueText = reader.readElementText();
+        if (valueText.isEmpty()) {
+            _lockStatus = SyncFileItem::LockStatus::UnlockedItem;
+        } else {
+            bool success = false;
+            _lockStatus = static_cast<SyncFileItem::LockStatus>(valueText.toInt(&success));
+            if (!success) {
```

Comment:
@mgallien Wouldn't it be more clear this way?
```
if (success) {
    _lockStatus = static_cast<SyncFileItem::LockStatus>(valueText.toInt(&success));
}
```

Reply:
that would change the behavior
`success` is modified by the call to `toInt` to know if that was a valid int or if the conversion failed